### ISSUE TITLE
'chmod' equivalent in snapshot preview1?

### ISSUE THREAD

**Issue by titzer** [WebAssembly/WASI#766]
It appears that snapshot preview 1 does not have a way to change file permissions. This is important functionality for one of my use cases (a compiler that generates executables). This functionality does appear to be planned for the next version of wasi.

Is snapshot preview1 being updated? Is there interest in adding a `chmod`-like capability?

If so, I would propose a simple `fs_chmod(fd: i32, mode: i32)` function.

**Comment by Timmmm** [comment]
Did this go anywhere? As far as I can see you can't even stat permissions either.

Some other issues mention `change-file-permissions-at`, e.g. #762 but that seems to not exist currently.

My use case is a WASI VFS backed by a Git repo. Git's only permission is a single executable flag.

**Comment by pchickey** [comment]
No, this was not taken further, wasi's filesystem has not gotten any new features since wasip1. In general, nobody has worked on expanding filesystem functionality because its not part of the production use cases for anyone who is paid to work on these standards and their implementations, so it would require a volunteer stepping up to do the work of championing additions to the standard and its implementations.

There was a broad set of additions to filesystem proposed as part of developing p2, but there was no labor available to do any of the implementation, so those additions were removed prior to launching p2.
